tests/test_mcmc_helper.py

```diff
@@ -43,27 +46,109 @@ def test_mcmc_helper_parameters():
 
 
 def test_mcmc_helper_target_function():
+    """ Set up priors on the model parameters such that we can
+    readily compute their expected values. """
     data = build_data()
     model = build_model(data)
 
+    prior_width = 200
+
     hmc_helper = gpflow.optimizers.SamplingHelper(
         model.trainable_parameters, model.log_marginal_likelihood
     )
+    target_log_prob_fn = hmc_helper.target_log_prob_fn
+
+    # Priors which are set on the constrained space
+    expected_log_prior = 0.
+    for param in model.trainable_parameters:
+        if param.value() < 1e-3:
+            param.assign(1.0)
+
+        param.transform = Exp()
+
+        low_value = -100
+        high_value = low_value + prior_width
+
+        param.prior = Uniform(low=np.float64(low_value), high=np.float64(high_value))
+
+        prior_density_on_constrained = 1 / prior_width
+        prior_density_on_unconstrained = prior_density_on_constrained * param.value()
+
+        expected_log_prior += np.log(prior_density_on_unconstrained)
+
+    log_likelihood = model.log_likelihood().numpy()
+    expected_log_prob = log_likelihood + expected_log_prior
+
+    hmc_helper = gpflow.optimizers.SamplingHelper(
+        model.trainable_parameters, model.log_marginal_likelihood
+    )
+
+    np.testing.assert_allclose(expected_log_prob, target_log_prob_fn())
+
+
+def test_mcmc_helper_target_function_unconstrained():
+    """ Verifies the objective for a set of priors which are defined on the unconstrained space.
+    """
+    data = build_data()
+    model = build_model(data)
+
+    # Set up priors on the model parameters such that we can readily compute their expected values.
+    expected_log_prior = 0.
+    prior_width = 200
+
+    hmc_helper = gpflow.optimizers.SamplingHelper(
+        model.trainable_parameters, model.log_marginal_likelihood
+    )
+
+    for param in model.trainable_parameters:
+        low_value = -100
+        high_value = low_value + prior_width
+        param.prior = Uniform(low=np.float64(low_value), high=np.float64(high_value))
+        param.prior_on = 'unconstrained'
+
+        prior_density = 1 / prior_width
+        expected_log_prior += np.log(prior_density)
 
     target_log_prob_fn = hmc_helper.target_log_prob_fn
+    expected_log_prob = model.log_likelihood().numpy() + expected_log_prior
+
+    assert np.isclose(expected_log_prob, target_log_prob_fn())
+
+
+def test_mcmc_helper_target_function_no_transforms():
```

Comment:
```suggestion
@pytest.mark.parametrize('prior_on', [gpflow.base.PriorOn.CONSTRAINED, gpflow.base.PriorOn.UNCONSTRAINED])
def test_mcmc_helper_target_function_no_transforms(prior_on):
```
should be the same regardless of where the prior is, right?